examples/squiral/main.go

```diff
@@ -0,0 +1,352 @@
+// Copyright 2019 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+// This demo is inspired by the xscreensaver 'squirals'.
+
+package main
+
+import (
+	"fmt"
+	"image/color"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+	"github.com/hajimehoshi/ebiten/inpututil"
+)
+
+type direction int
+
+const (
+	width  = 800
+	height = 600
+	scale  = 2
+	amount = width / 32
+)
+
+const (
+	east direction = iota
+	south
+	west
+	north
+)
+
+type palette struct {
+	name   string
+	colors []color.Color
+}
+
+var (
+	background = color.Black
+
+	palettes = []palette{
+		{
+			name: "sand dunes",
+			colors: []color.Color{
+				color.RGBA{0xF2, 0x74, 0x05, 0xFF}, //#F27405
```

Comment:
Actually I would love to keep them because my editor highlights these with actual colors. See screenshot:

![colors](https://user-images.githubusercontent.com/791374/66338105-bb3c9800-e940-11e9-9a5e-b5320de2e196.png)

Reply:
Hmm, I tend not to want to leave comments only for specific editors...

Reply:
There are plugins for nearly every existing editor to highlight colors in hex notation. What is so bad about these comments? It also enables copy&paste of the colors. Is there a go std function that takes a hex color to create a `color.Color`? I didn't find one when I implemented this.

Reply:
> There are plugins for nearly every existing editor to highlight colors in hex notation.

VSCode, or other editors? I'm not sure how common this is.

> What is so bad about these comments?

The comments don't add any additional information.

>  Is there a go std function that takes a hex color to create a color.Color?

Unfortunately there is no such function. You can make this, but in this case, the function would be like `hexToColor(0xf27405)`, and this would not be what you wanted.

Reply:
> > There are plugins for nearly every existing editor to highlight colors in hex notation.
> 
> VSCode, or other editors? I'm not sure how common this is.

VSCode and other editors. It is very common for web developers. 


> > What is so bad about these comments?
> 
> The comments don't add any additional information.

Well depends.. only if the editor supports it. I understand you would not want these comments in a library e.g. but in a demo I don't see the harm.

> > Is there a go std function that takes a hex color to create a color.Color?
> 
> Unfortunately there is no such function. You can make this, but in this case, the function would be like `hexToColor(0xf27405)`, and this would not be what you wanted.

The option would be a function that consumes strings of the format `#rrggbb` but I am reluctant to write it just to get rid of a few comments :)